src/main/java/org/myrobotlab/codec/ApiMessages.java

```diff
@@ -153,16 +107,18 @@ public Object process(MessageSender sender, OutputStream out, Message requestUri
       MethodCache.cache(clazz, method);
 
     } else {
+      // FIXME - this is terrible ! .. changing to a different api ??? wtf ???
       // First GET /api/messages - has data == null !
       // use different api to process GET ?
       // return hello ?
       // FALLBACK
-      ApiFactory api = ApiFactory.getInstance();
-      String newUri = requestUri.uri.replace("/messages", "/service");
+      apiKey = "service";

```

Comment:
The assignment of apiKey here seems like a side effect ,  not sure I understand the concern or detail of what's going on,  so I guess it's good to have the FIXME / TODO in the code for cleaning this up.

Reply:
Ya.  There is more cleanup needed in Api - currently working on this